chemprop/train/loss_functions.py

```diff
@@ -171,16 +208,189 @@ def wasserstein_loss(model_spectra: torch.tensor, target_spectra: torch.tensor,
     model_spectra = torch.where(mask, model_spectra, zero_sub)
     sum_model_spectra = torch.sum(model_spectra, axis=1, keepdim=True)
     model_spectra = torch.div(model_spectra, sum_model_spectra)
 
     # Calculate loss value
-    target_cum = torch.cumsum(target_spectra,axis=1).to(torch_device)
-    model_cum = torch.cumsum(model_spectra,axis=1).to(torch_device)
+    target_cum = torch.cumsum(target_spectra, axis=1)
+    model_cum = torch.cumsum(model_spectra, axis=1)
     loss = torch.abs(target_cum - model_cum)
-    loss = loss.to(torch_device)
+
+    return loss
+
+
+def normal_mve(pred_values, targets):
+    """
+    Use the negative log likelihood function of a normal distribution as a loss function used for making
+    simultaneous predictions of the mean and error distribution variance simultaneously.
+
+    :param pred_values: Combined predictions of means and variances of shape(data, tasks*2).
+                        Means are first in dimension 1, followed by variances.
+    :return: A tensor loss value.
+    """
+    # Unpack combined prediction values
+    pred_means, pred_var = torch.split(pred_values, pred_values.shape[1] // 2, dim=1)
+
+    return torch.log(2 * np.pi * pred_var) / 2 + (pred_means - targets) ** 2 / (
+        2 * pred_var
+    )
+
+
+# evidential classification
+def dirichlet_class_loss(alphas, target_labels, lam=0):
+    """
+    Use Evidential Learning Dirichlet loss from Sensoy et al in classification datasets.
+    :param alphas: Predicted parameters for Dirichlet in shape(datapoints, tasks*2).
+                   Negative class first then positive class in dimension 1.
+    :param target_labels: Digital labels to predict in shape(datapoints, tasks).
+    :lambda: coefficient to weight KL term
+
+    :return: Loss
+    """
+    torch_device = alphas.device
+    num_tasks = target_labels.shape[1]
+    num_classes = 2
+    alphas = torch.reshape(alphas, (alphas.shape[0], num_tasks, num_classes))
+
+    y_one_hot = torch.eye(num_classes, device=torch_device)[target_labels.long()]
+
+    # SOS term
+    S = torch.sum(alphas, dim=-1, keepdim=True)
+    p = alphas / S
+    A = torch.sum(torch.pow((y_one_hot - p), 2), dim=-1, keepdim=True)
+    B = torch.sum((p * (1 - p)) / (S + 1), dim=-1, keepdim=True)
+    SOS = A + B
+
+    # KL
+    alpha_hat = y_one_hot + (1 - y_one_hot) * alphas
+
+    beta = torch.ones_like(alpha_hat)
+    S_alpha = torch.sum(alpha_hat, dim=-1, keepdim=True)
+    S_beta = torch.sum(beta, dim=-1, keepdim=True)
+
+    ln_alpha = torch.lgamma(S_alpha) - torch.sum(
+        torch.lgamma(alpha_hat), dim=-1, keepdim=True
+    )
+    ln_beta = torch.sum(torch.lgamma(beta), dim=-1, keepdim=True) - torch.lgamma(
+        S_beta
+    )
+
+    # digamma terms
+    dg_alpha = torch.digamma(alpha_hat)
+    dg_S_alpha = torch.digamma(S_alpha)
+
+    # KL
+    KL = (
+        ln_alpha
+        + ln_beta
+        + torch.sum((alpha_hat - beta) * (dg_alpha - dg_S_alpha), dim=-1, keepdim=True)
+    )
```

Comment:
make this a module function because the code is repeated in the loss function below